subworkflows/local/mtx_conversion.nf

```diff
@@ -1,62 +1,86 @@
-        // TODO: Adapt cellranger-arc subworkflow like cellranger to remove this snippet here.
-        if (params.aligner in [ 'cellrangerarc' ]) {
-            mtx_matrices = mtx_matrices.map { meta, mtx_files ->
-                    [ meta, mtx_files.findAll { it.toString().contains("filtered_feature_bc_matrix") } ]
-                }
-                .filter { meta, mtx_files -> mtx_files } // Remove any that are missing the relevant files
-        }
+        ch_h5ads    = Channel.empty()
 
         //
-        // Convert matrix to h5ad
+        // MODULE: Convert matrices to h5ad
         //
         MTX_TO_H5AD (
             mtx_matrices,
             txp2gene,
             star_index
         )
+        ch_versions = ch_versions.mix(MTX_TO_H5AD.out.versions.first())
+
+        // fix channel size when kallisto non-standard workflow
+        if (params.aligner == 'kallisto' && !(params.kb_workflow == 'standard')) {
+            ch_h5ads =
+            MTX_TO_H5AD.out.h5ad
+                .transpose()
+                .map { meta, h5ad ->
+                    def meta_clone = meta.clone()
+                    def spc_prefix = h5ad.toString().contains('unspliced') ? 'un' : ''
+
+                    meta_clone["input_type"] = "${meta.input_type}_${spc_prefix}spliced"
+
+                    [ meta_clone, h5ad ]
+                }
+        } else {
+            ch_h5ads = MTX_TO_H5AD.out.h5ad
+        }
+
+        //
+        // SUBWORKFLOW: Run cellbender emptydrops filter
+        //
+        if ( !params.skip_emptydrops && !(params.aligner in ['cellrangerarc']) ) {
+
+            // emptydrops should only run on the raw matrices thus, filter-out the filtered result of the aligners that can produce it
+            EMPTY_DROPLET_REMOVAL (
+                ch_h5ads.filter { meta, mtx_files -> meta.input_type.contains('raw') }
+            )
+            ch_h5ads = ch_h5ads.mix( EMPTY_DROPLET_REMOVAL.out.h5ad )
+
+        }
+
+        //
+        // MODULE: Convert to Rds with AnndataR package
+        //
+        ANNDATAR_CONVERT (
+            ch_h5ads
+        )
 
         //
         // Concat sample-specific h5ad in one
         //
-        ch_concat_h5ad_input = MTX_TO_H5AD.out.h5ad.groupTuple() // gather all sample-specific files / per type
+        ch_concat_h5ad_input = ch_h5ads.groupTuple() // gather all sample-specific files / per type
         if (params.aligner == 'kallisto' && params.kb_workflow != 'standard') {
             // when having spliced / unspliced matrices, the collected tuple has two levels ( [[mtx_1, mtx_2]] )
             // which nextflow break because it is not a valid 'path' thus, we have to remove one level
             // making it as [ mtx_1, mtx_2 ]
             ch_concat_h5ad_input = ch_concat_h5ad_input.map{ type, matrices -> [ type, matrices.flatten().toList() ] }
         }
+
         CONCAT_H5AD (
             ch_concat_h5ad_input,
             samplesheet
         )
```

Comment:
The concatenated h5ad could then be converted to Seurat/Bioconductor again as well for convenience

Reply:
Yes, that is true :)

Reply:
made the inclusion
😄